nix/overlays/haskell-packages.nix

```diff
@@ -59,6 +59,15 @@ let
           }
           { });
 
+      aeson =
+        lib.dontCheck (prev.callHackageDirect
+          {
+            pkg = "aeson";
+            ver = "2.0.3";
+            sha256 = "1y3jj4bhviss6ff9lwxki0zbdcl1rb398bk4s80zvfpnpy7p94cx";
+          }
+          { });
```

Comment:
I think I naively copied the warp entry correctly (with a wrong hash), but it results in an "infinite recursion"...

Reply:
Ah, seems the problem is that `callHackageDirect` depends on `cabal2nix`, which in turn depends on aeson.

https://github.com/NixOS/nixpkgs/issues/83098

Reply:
ah that makes sense, thanks!

Reply:
fixed this one by adding an extra cabal2nix overlay